felix/bpf-gpl/globals.h

```diff
@@ -25,19 +25,23 @@ struct name {				\
 	__u8 iface_name[16];		\
 	__u32 log_filter_jmp;		\
 	__u32 jumps[40];		\
+}
+
+DECLARE_TC_GLOBAL_DATA(cali_tc_global_data, ipv6_addr_t);
+struct cali_tc_globals {
+	struct cali_tc_global_data data;
+
 	/* Needs to be 32bit aligned as it is followed by scratch area for 				\
 	 * building headers. We reuse the same slot in state map to save 				\
 	 * ourselves a lookup. 										\
 	 */												\
 	__u32 __scratch[]; /* N.B. this provides pointer to the location but does not add to the size */ \
-}
+};
 
-DECLARE_TC_GLOBALS(cali_tc_globals, ipv46_addr_t);
-/* cali_tc_globals_v6 is for userspace as cali_tc_globals are used for ipv4 in
- * userspace, but it has the exact same layout as cali_tc_globals in eBPF when
- * compiled for ipv6.
- */
-DECLARE_TC_GLOBALS(cali_tc_globals_v6, ipv6_addr_t);
+struct cali_globals {
```

Comment:
what about `tc_preamble_globals` , `tc_preamble_confic` or something like that. It is tc and preamble specific. There are also xdp and ctlb globals.